Doc/reference/datamodel.rst

```diff
@@ -2947,6 +2950,9 @@ through the object's keys; for sequences, it should iterate through the values.
    :meth:`~object.__bool__` method and whose :meth:`!__len__` method returns zero is
    considered to be false in a Boolean context.
 
+   Two successive calls to :meth:`!__len__` on the same object must
+   return same value.
```

Comment:
```suggestion
   Note: Python interpreters are permitted to assume that two successive calls
   to  :meth:`!__len__` on the same object in the same thread will return the same
   value. While this expected invariant is not explicitly enforced, failing to abide
   by it will result in implementation dependent runtime behaviour. This
   implementation dependent behaviour may also be encountered when mutable
   objects are shared across threads without appropriate synchronization.
```

Reply:
Here is a typo (``__len__`` -> ``__bool__``).  But if we are going with this lengthly wording - I think it's better just point to the ``__bool__`` docs.